akka-aeron/src/main/scala/akka/aeron/AeronExtension.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.aeron
+
+import akka.actor.{ ExtendedActorSystem, Extension, ExtensionId, ExtensionIdProvider }
+import akka.annotation.InternalApi
+import akka.event.{ Logging, LoggingAdapter }
+import com.typesafe.config.Config
+import io.aeron.{ Aeron, AvailableImageHandler, Image, UnavailableImageHandler }
+
+sealed trait MediaDriverLocation
+case object Embedded extends MediaDriverLocation
+final case class External(location: String) extends MediaDriverLocation
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object AeronSettings {
+
+  def apply(config: Config): AeronSettings = {
+    new AeronSettings(
+      config.getDuration("connection.driver-timeout").toMillis,
+      config.getString("connection.aeron-directory") match {
```

Comment:
Not a personal fan of config work outside a settings class, mixed in with the regular code

Reply:
I usually like to have the settings class not be aware of a `Config` so they are easy to construct for tests and then have the reading from a `Config` in an apply. Do you see any benefits of putting the config work inside the settings class?